upstairs/src/downstairs.rs

```diff
@@ -2706,17 +2708,21 @@ impl Downstairs {
     /// The live-repair may continue after this point to clean up reserved jobs,
     /// to avoid blocking dependencies, but jobs are replaced with no-ops.
     fn abort_repair(&mut self, up_state: &UpstairsState) {
-        assert!(self.clients.iter().any(|c| {
-            c.state() == DsState::LiveRepair ||
+        assert!(self.clients.iter().any(|c| matches!(
+            c.state(),
+            DsState::LiveRepair |
                 // If connection aborted, and restarted, then the re-negotiation
                 // could have won this race, and transitioned the reconnecting
                 // downstairs from LiveRepair to Faulted to LiveRepairReady.
-                c.state() == DsState::LiveRepairReady ||
+                DsState::LiveRepairReady |
                 // If just a single IO reported failure, we will fault this
                 // downstairs and it won't yet have had a chance to move back
                 // around to LiveRepairReady yet.
-                c.state() == DsState::Faulted
-        }));
+                DsState::Faulted |
+                // It's also possible for a Downstairs to be in the process of
+                // stopping, due a fault or disconnection
+                DsState::Stopping(..) // XXX should we be more specific here?
```

Comment:
maybe?

also, slight nit: I think we should do away with this assert, move the comments to the match on client state below, and panic on the catch-all case. We're missing both Faulted and Stopping(..) enum variants in the below match, and I'd rather be explicit there that we don't have to do anything in those cases.